src/System.Private.ServiceModel/src/System/ServiceModel/Security/SecuritySessionClientSettings.cs

```diff
@@ -487,7 +487,7 @@ protected internal override async Task OnOpenAsync(TimeSpan timeout)
             {
                 TimeoutHelper timeoutHelper = new TimeoutHelper(timeout);
                 SetupSessionTokenProvider();
-                SecurityUtils.OpenTokenProviderIfRequired(_sessionTokenProvider, timeoutHelper.RemainingTime());
+                await SecurityUtils.OpenTokenProviderIfRequiredAsync(_sessionTokenProvider, timeoutHelper.RemainingTime());
```

Comment:
should you have .ConfigureAwait(false) on these?

Reply:
It's not needed. I hated the idea of littering my code everywhere with ConfigureAwait because it's messy, and it only takes missing one to create a difficult to find bug. So instead I have some code which is called when calling WCF using an async contract which checks if you are using a non-default sync context or a non-default task scheduler, and if you are, forces an artificial continuation into the worker thread pool. So this code is guaranteed to not need ConfigureAwait. ConfigureAwait is needed if you _might_ go async. If you are guaranteed to always go async (and WCF always does as it always does network IO which requires a response), then you can just forcibly go async from the start.

Reply:
OK, cool just checking.
If you have a controlled call graph, you are probably OK,
We created a Roslyn Code Analyzer to check for our other libraries we have many entry points.

@mconnew What was the memory leak?

Reply:
The memory leak was not closing the channel. The channel factory will hold on to the channel in an internal list. Each call would create a new instance which would accumulate instances. The fix is to close the channel after usage. Also we now close the channel factory when the parent channel factory is closed as that closes the token provider.